sklearn/manifold/spectral_embedding_.py

```diff
@@ -520,3 +529,42 @@ def fit_transform(self, X, y=None):
         """
         self.fit(X)
         return self.embedding_
+
+    def transform(self, X):
+        """
+        Transform new points into embedding space.
+        Parameters
+        ----------
+        X : array-like, shape = (n_samples, n_features)
+            Vector, where n_samples is the number of samples
+            and n_features is the number of features.
+
+        Returns
+        -------
+        X_new : array, shape = (n_samples, n_components)
+        """
+        X = check_array(X)
+
+        if (self.affinity == "nearest_neighbors"):
+            d = pairwise_distances(X, self.training_data_, n_jobs=self.n_jobs,
+                                   metric="euclidean")
+            d[(d == 0)] = np.inf
+            M = np.zeros(d.shape)
```

Comment:
one-letter names (for example "M", "d") are bad names for variables
(dd is also not a self-commented name)